src/components/TextField/TextField.tsx

```diff
@@ -38,50 +74,89 @@ export class TextField extends React.Component<TextFieldProps> {
     this.TextFieldRef.current.blur();
   };
 
-  getSuffix = () => {
+  _getSuffix = () => {
     const {
+      error,
       errorMessage,
-      success = false,
-      successIcon = false,
-      error = false,
+      success,
+      successIcon,
       suffix,
+      withClearButton,
+      clearButtonAriaLabel,
+      clearButtonAriaLabelledby,
+      onClear,
+      value,
+      disabled,
     } = this.props;
-    let suffixToShow = suffix;
-
-    if (errorMessage && error) {
-      suffixToShow = (
-        <Tooltip
-          appendTo="scrollParent"
-          placement="top-end"
-          skin={TooltipSkin.Error}
-          content={errorMessage}
-          moveBy={{ x: 5, y: 0 }}
-        >
-          <ErrorIcon />
-        </Tooltip>
-      );
-    }
-
-    if (successIcon && success) {
-      suffixToShow = <SuccessIcon data-hook="successIcon" />;
-    }
-    return suffixToShow;
+
+    const shouldShowCustomSuffix = !!suffix;
+    const shouldShowErrorIcon = error && errorMessage;
+    const shouldShowSuccessIcon = successIcon && success;
+    const shouldShowClearButton = withClearButton && value && !disabled;
+
+    const hasSuffix =
+      shouldShowCustomSuffix ||
+      shouldShowErrorIcon ||
+      shouldShowSuccessIcon ||
+      shouldShowClearButton;
+
+    const shouldAddLeftPaddingToCustomSuffix =
+      !shouldShowErrorIcon && !shouldShowSuccessIcon && shouldShowClearButton;
+
+    return hasSuffix ? (
+      <div className={classes.suffixWrapper}>
+        <div className={classes.gap} />
+        {shouldShowClearButton && (
+          <IconButton
+            className={classes.clearButton}
+            data-hook={DATA_HOOKS.CLEAR_BUTTON}
+            aria-label={clearButtonAriaLabel}
+            aria-labelledby={clearButtonAriaLabelledby}
+            onClick={() => onClear && this.props.onClear()}
```

Comment:
I would do
```suggestion
            onClick={onClear ? () => this.props.onClear() : undefined}
```
cause there's no need to pass a function at all if the callback doesn't exist